CRM/Utils/System/Joomla.php

```diff
@@ -464,11 +464,18 @@ public function permissionDenied() {
     throw new CRM_Core_Exception(ts('You do not have permission to access this page.'));
   }
 
+  /**
+   * Start a new session.
+   */
+  public function sessionStart() {
+    JFactory::getSession()->start();
```

Comment:
In the interests of full disclosure... I was poking around [dev/joomla#21](https://lab.civicrm.org/dev/joomla/-/issues/21), wherein session-mangement for `bin/cron.php` (etal) on Joomla gets bungled and fails. It's an inter-related topic.

This bit here actually causes `bin/cron.php` to fail in a different way. Specifically, changing from `session_start()` to `JFactory::getSession()->start()` means that you need to have Joomla available. But in the call-paths of `bin/cron.php`, Joomla isn't available at this point (ie it calls `$userSystem->sessionStart()` before `$userSystem->loadBootStrap()`).

This alternate branch has an additional 1-line commit which (taken together) fixes dev/joomla#21 for me.  However, it *might* break `extern/soap.php` specifically (because, AFAIK, that's the only stateful end-point in `bin/` or `extern/`).

https://github.com/totten/civicrm-core/commits/master-joomla-session-b